### ISSUE TITLE
Excessive disk writes after enabling CPUManager

### ISSUE THREAD

**Issue by SaveTheRbtz** [kubernetes/kubernetes#103006]
#### What happened:
We've recently enabled CPUManager and TopologyManagers:
```
CPUManagerPolicy:      "static",
TopologyManagerPolicy: kubelet_config_v1.BestEffortTopologyManagerPolicy,
TopologyManagerScope:  kubelet_config_v1.ContainerTopologyManagerScope,
```

This lead to major increase in our SSD write rates, that negatively affects their lifetime.  Nodes that have CPU Manager enabled have around 6x the disk IO than ones without the manager:
![image](https://user-images.githubusercontent.com/169976/122610917-42f30900-d035-11eb-9dd2-549864cb1d86.png)

Seems like the excessive io comes from `countainerd` database writes (`/var/lib/containerd/io.containerd.metadata.v1.bolt/meta.db`):
```
$ ext4slower 0 | fgrep W
19:56:02 containerd     106737 W 4096    3740        0.00 meta.db
19:56:02 containerd     106737 W 4096    3800        0.00 meta.db
19:56:02 containerd     106737 W 4096    3844        0.00 meta.db
...
```

These are likely due to periodic `containerd` `UpdateContainerResources` requests:
```
$ journalctl -f -u containerd | fgrep 60e47a60ef8c3b31b877aa6d350156b6e8edcde94624eb67cc6673c267f0cf1c
Jun 18 19:57:15 sjc8d-rl15-15a containerd[106737]: time="2021-06-18T19:57:15.208439833Z" level=info msg="UpdateContainerResources for \"60e47a60ef8c3b31b877aa6d350156b6e8edcde94624eb67cc6673c267f0cf1c\" with &LinuxContainerResources{CpuPeriod:0,CpuQuota:0,CpuShares:0,MemoryLimitInBytes:0,OomScoreAdj:0,CpusetCpus:0-1,14-49,62-95,CpusetMems:,HugepageLimits:[]*HugepageLimit{},}"
...
Jun 18 19:57:26 sjc8d-rl15-15a containerd[106737]: time="2021-06-18T19:57:26.460660552Z" level=info msg="UpdateContainerResources for \"60e47a60ef8c3b31b877aa6d350156b6e8edcde94624eb67cc6673c267f0cf1c\" with &LinuxContainerResources{CpuPeriod:0,CpuQuota:0,CpuShares:0,MemoryLimitInBytes:0,OomScoreAdj:0,CpusetCpus:0-1,14-49,62-95,CpusetMems:,HugepageLimits:[]*HugepageLimit{},}"
```

Looking at these, seems like `LinuxContainerResources` is not changing.  Is it possible to make these requests only if there is a change, to reduce amount of steady state writes to the drives?

#### Environment:

Node:
```
NAME              STATUS   ROLES    AGE     VERSION   INTERNAL-IP    EXTERNAL-IP   OS-IMAGE             KERNEL-VERSION     CONTAINER-RUNTIME
sjc8d-rl15-15a    Ready    <none>   20h     v1.20.1   10.60.158.40   <none>        Ubuntu 20.04.2 LTS   5.8.0-49-generic   containerd://1.5.2
```

k8s:
```
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.1", GitCommit:"c4d752765b3bbac2237bf87cf0b1c2e307844666", GitTreeState:"clean", BuildDate:"2020-12-18T12:09:25Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.1", GitCommit:"c4d752765b3bbac2237bf87cf0b1c2e307844666", GitTreeState:"clean", BuildDate:"2020-12-18T12:00:47Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by fromanirh** [comment]
yes, enabling the resource manager creates more writes because the on-disk checkpoint files need to be updated. For CPU Manager, recently this optimization was merged and willbe featured on 1.22: https://github.com/kubernetes/kubernetes/commit/664603948114d2c87df479d297fe788ab10eba06

**Comment by SaveTheRbtz** [comment]
Closing since this was allegedly fixed in 1.22.  Will reopen if the problem persists after our 1.21->1.22 upgrade.

/close
